01_decouverte_micropython/01_decouverte_micropython.md

```diff
@@ -299,6 +299,75 @@ Modifie ta barre de chargement pour qu'au lieu d'allumer les diodes d'un coup
 (avec une intensité de 9), elles s'allument en douceur (en faisant
 progressivement varier l'intensité de 0 à 9).
 
+## L'indentation
+
+À partir de maintenant, tes programmes commenceront à être plus compliqués. Il
+est tout de même nécessaire de comprendre une partie importante de la syntaxe
+de Python : l'**indentation**.
+
+L'indentation correspond au nombre d'espaces avant le début d'une ligne. Par
+exemple, la première ligne ci-dessous à une indentation égale à **0** (pas
+d'espaces en début de ligne), et la deuxième ligne à une indentation égale à
+**4** (quatre espaces en début de ligne).
+
+```python
+for i in range(5):
+    display.scroll(str(i))
+```
+
+Cette indentation permet à Python de savoir, par exemple, quelles parties d'un
+programme sont à répéter. Comparons deux exemples.
+
+Dans l'exemple ci-dessous, la série suivante de chaînes de caractères est
+affichée :
+- 0
+- end
+- 1
+- end
+
+```python
+for i in range(2):
+    display.scroll(str(i))
+    display.scroll("end")
+```
+
+Dans l'exemple ci-dessous, la série suivante de chaînes de caractères est
+affichée :
+- 0
+- 1
+- end
+
+```python
+for i in range(2):
+    display.scroll(str(i))
+display.scroll("end")
+```
+
+La dernière ligne, `display.scroll("end")`, est en effet "en dehors" de la
+boucle `for`, et s'exécute donc une seule fois.
```

Comment:
est-ce que ce serait pas mieux d'introduire la notion de scope / bloc d'exécution pour mieux montrer cette partie là ? Ça rendra un peu plus explicite le principe.

Reply:
Pourquoi pas, mais j'ai peur que ça fasse beaucoup car avec Python il faudrait expliquer les scopes et l'indentation en même temps. Et puis les scopes sont un peu bizarres en Python, genre tu peux faire

```py
if foo:
  bar = 2
else:
  bar = 3

print(bar)
```

et c'est valide même si `bar` n'est pas utilisée avant, contrairement à plein de langages.

Reply:
Ouais évidemment, mais on est peut-être pas obligé d'aller dans ces détails là (qui vont probablement plus mener à de la confusion qu'autre chose).

Je pense qu'on peut juste se contenter de faire une grosse simplification scope/indentation et juste dire que ça permet de "découper" en sortes de bloc d'exécution qui forment des sortes de "sous-parties" d'une fonction. 
Mon avis étant, appelé plus explicitement peut - je pense - faciliter la compréhension. 

Après on peut faire au fur-et à mesure et commencer par parler comme t'as fait de la partie boucle for, puis après généraliser, etc.

Évidemment c'est une proposition qui se discute, le but étant de faire le plus explicite possible pour les gens concernés par le TP :).

Reply:
Du coup je vais push as-is vu que j'ai d'autres chats à fouetter, mais j'essaierai de trouver des améliorations par rapport à ça.